packages/apollo/src/updaters/removeFromList.ts

```diff
@@ -26,10 +24,19 @@ export default function removeFromList<Result, Vars = {}>(
       }
     }
 
-    set(
-      nextResult,
+    const rootItem = listPath.split('.')[0];
+    // @ts-ignore
+    const resultRoot = queryResult[rootItem] as StoreObject;
```

Comment:
Not sure if there's a better way to identify nested nodes but to evict cache we need at least the top level node ID for the `cache.identify` call below.